xt_FULLCONENAT.c

```diff
@@ -590,9 +590,15 @@ static unsigned int fullconenat_tg(struct sk_buff *skb, const struct xt_action_p
       }
     }
 
-    new_ip = get_device_ip(skb->dev);
-    newrange.min_addr.ip = new_ip;
-    newrange.max_addr.ip = new_ip;
+    if(mr->range[0].flags & NF_NAT_RANGE_MAP_IPS) {
+      newrange.min_addr.ip = mr->range[0].min_ip;
+      newrange.max_addr.ip = mr->range[0].max_ip;
+    }
+    else {
```

Comment:
Though it isn't a big deal, would you mind putting the `else` statement into one line (i.e. `} else {`) to better match the coding style for kernel programming? Anyway thanks for your contribution. This patch looks good.